addons/web/static/tests/core/utils/hooks.test.js

```diff
@@ -0,0 +1,587 @@
+        expect(document.activeElement).toBe(component.inputRef.el);
+
+        component.render();
+        await animationFrame();
+        expect(document.activeElement).toBe(component.inputRef.el);
+    });
+
+    test`desktop`("conditional autofocus", async () => {
+        class MyComponent extends Component {
+            static props = ["*"];
+            static template = xml`
+                <span>
+                    <input t-if="showInput" type="text" t-ref="autofocus" />
+                </span>
+            `;
+            setup() {
+                this.inputRef = useAutofocus();
+                this.showInput = true;
+            }
+        }
+
+        const component = await mountWithCleanup(MyComponent);
+        expect(document.activeElement).toBe(component.inputRef.el);
+
+        component.showInput = false;
+        component.render();
+        await animationFrame();
+        expect(document.activeElement).toBe(document.body);
+
+        component.showInput = true;
+        component.render();
+        await animationFrame();
+        expect(document.activeElement).toBe(component.inputRef.el);
+    });
+
+    test("returns also a ref when screen has touch", async () => {
+        expect(1);
+        class MyComponent extends Component {
+            static template = xml`
+                <span>
+                    <input type="text" t-ref="autofocus" />
+                </span>
+            `;
+            static props = ["*"];
+            setup() {
+                this.inputRef = useAutofocus();
+                onMounted(() => {
+                    // There is a ref, but the input is not focused
+                    expect(this.inputRef.el).toBeTruthy();
+                });
+            }
+        }
+
+        patchWithCleanup(browser, {
+            matchMedia: (media) => {
+                if (media === "(pointer:coarse)") {
+                    return { matches: true };
+                }
+                this._super();
+            },
+        });
+
+        await mountWithCleanup(MyComponent);
+    });
+
+    test("works when screen has touch and you provide mobile param", async () => {
+        class MyComponent extends Component {
+            static props = ["*"];
+            static template = xml`
+                <span>
+                    <input type="text" t-ref="autofocus" />
+                </span>
+            `;
+            setup() {
+                this.inputRef = useAutofocus({ mobile: true });
+            }
+        }
+
+        patchWithCleanup(browser, {
+            matchMedia: (media) => {
+                if (media === "(pointer:coarse)") {
+                    return { matches: true };
+                }
+                this._super();
+            },
+        });
+
+        const component = await mountWithCleanup(MyComponent);
+        expect(document.activeElement).toBe(component.inputRef.el);
+    });
+
+    test("does not focus when screen has touch", async () => {
```

Comment:
ok so the missing assertion of "returns also a ref when screen has touch" is here. Those two tests should probably have been merged